opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -0,0 +1,677 @@
+            'probability' is not required for training purpose.
+        output_model_path: str=None
+            the path to store trained custom model. If None, default as current folder path
+        output_model_name: str=None
+            the name of the trained custom model. If None, default as 'trained_model.pt'
+        zip_file_name: str =None
+            Optional, file name for zip file. if None, default as custom_tasb_model.zip
+        use_accelerate: bool = False,
+            Optional, use accelerate to fine tune model. Default as false to not use accelerator.
+        compute_environment: str
+            optional, compute environment type to run model, if None, default using 'LOCAL_MACHINE'
+        num_machines: int
+            optional, number of machine to run model , if None, default using 1
+        num_processes: int
+            optional, number of processors to run model , if None, default using 1
+        learning_rate: float
+            optional, learning rate to train model, default is 2e-5
+        Return:
+            None
+        """
+
+        query_df = self.read_queries(read_path)
+
+        train_examples = self.load_sentence_transformer_example(
+            query_df, use_accelerate
+        )
+
+        # if use accelerator to train model, run auto setuo accelerate confi and launch train_model function
+        # with the number of processors provided by users
+        # if NOT use accelerator, trigger train_model function with default setting
+
+        if use_accelerate is True:
+            self.set_up_accelerate_config(
+                compute_environment=compute_environment,
+                num_machines=num_machines,
+                num_processes=num_processes,
+            )
+
+            if self.is_notebook():
+                notebook_launcher(
+                    self.train_model,
+                    args=(
+                        train_examples,
+                        self.model_url,
+                        output_model_path,
+                        output_model_name,
+                        use_accelerate,
+                        learning_rate,
+                    ),
+                    num_processes=num_processes,
+                )
+            else:
+                subprocess.run(
+                    [
+                        "accelerate launch self.train_model(train_examples, self.model_url, output_model_path, output_model_name,use_accelerate,learning_rate)"
+                    ]
+                )
+        else:
+            self.train_model(
+                train_examples,
+                self.model_url,
+                output_model_path,
+                output_model_name,
+                use_accelerate,
+                learning_rate,
+            )
+
+        self.zip_model(output_model_path, zip_file_name)
+        return None
+
+    #     helpful functions:
+    def read_queries(self, read_path: str = None) -> pd.DataFrame:
```

Comment:
Should this method only used internally in this class? Based on https://peps.python.org/pep-0008/#method-names-and-instance-variables
> Use one leading underscore only for non-public methods and instance variables.

Reply:
Aligned to keep the function public but will add step by step training usage in the documentations